exercises/lamar_moore/ex02_oo_basics/src/ShapeSorter.cpp

```diff
@@ -0,0 +1,172 @@
+#include <iostream>
+#include <algorithm>
+#include <memory>
+#include "ShapeSorter.h"
+
+
+using namespace std;
+
+namespace Shapes
+{
+	/** Functor which is used to compare shape types with a matching value in std::remove_if
+	*/
+	class MatchTypeFunctor
```

Comment:
Excellent solution!

Two minor points.
1. put these definitions into an anonymous namespace block.
1. passing a shared pointer by value implies change of ownership and can be costly in a tight-loop due to internal increment/decrement of the copies of the shared_ptr. Would be better to pass by `const shared_ptr<Shape>::element_type&` I'd be happy to explain this. But it's not mandatory for you to change this.